### PR TITLE
Added a new resource for the C Programming language

### LINKED ISSUE #981: [Add] C Programming Resource
**Site URL**
https://www.freecodecamp.org/
https://www.freecodecamp.org/news/learn-c-programming-classic-book-dr-chuck/

**Description of the above provided site**
FreeCodeCamp offers a comprehensive and free curriculum that covers various programming languages, including C. Their C programming course provides interactive lessons, coding challenges, and projects to help learners develop their C programming skills effectively.

**Category**
Languages

**Sub Category**
C Programming

**Checklist:**
- [x] I have checked the existing issues.
- [x] I have read the Contributing Guidelines.
- [x] I am willing to work on this issue (optional).
- [x] I am a GSSoC'23 contributor.

### PR DESCRIPTION
**Pull request by sakshimishra19**
## Fixes Issue

Closes #981 

## Changes proposed

Added a new resource for the C Programming language

### CODE REVIEW

File: database/languages/c-programming.json [hunk 1]
```diff
@@ -1 +1,7 @@
-[]
\ No newline at end of file
+[ {
+    "name": "FreeCodeCamp",
```
Comment (CBID2) [inline database/languages/c-programming.json:2]: [Nitpick] We have a few sources from FreecodeCamp, so I think it's best to use the course's name to make it more distinct from the others.
```suggestion
    "name": "FreeCodeCamp",
```

**Review by CBID2 — CHANGES_REQUESTED** [review]
Hi @sakshimishra19! :) I just gave a suggestion on how to improve the source's name. 😀

File: database/languages/c-programming.json (hunk 1 above)
Comment (sakshimishra19) [reply database/languages/c-programming.json]: Hey @CBID2 
I have changed the source name from "FreeCodeCamp" to Course-name

.
Kindly review it :)

File: database/languages/c-programming.json [hunk 2]
```diff
@@ -1 +1,7 @@
-[]
\ No newline at end of file
+[ {
+    "name": "Learn C Programming Using the Book by Kernighan and Ritchi",
+    "description": "FreeCodeCamp offers a free C programming course with interactive lessons and coding challenges.",
```
Comment (CBID2) [inline database/languages/c-programming.json:3]: [Proofreading]
```suggestion
    "description": "This free C programming course offers interactive lessons and coding challenges to help you learn this language.",
```

**Review by CBID2 — CHANGES_REQUESTED** [review]
Just posted a suggestion on how to improve the description @sakshimishra19! :) Keep it up! :) 👍🏾

**Comment by sakshimishra19** [comment]
Hey @CBID2 
The description has been changed.

Kindly review it

**Review by CBID2 — APPROVED** [review]
LGTM! :)

**Comment by rupali-codes** [comment]
@sakshimishra19 could you please merge conflicts?

File: database/languages/c-programming.json [hunk 3]
```diff
@@ -1 +1,7 @@
-[]
\ No newline at end of file
+[ {
+    "name": "Learn C Programming Using the Book by Kernighan and Ritchi",
```
Comment (rupali-codes) [inline database/languages/c-programming.json:2]:
```suggestion
    "name": "Learn C Programming",
```

File: database/languages/c-programming.json [hunk 4]
```diff
@@ -1 +1,7 @@
...
+    "description": "This free C programming course offers interactive lessons and coding challenges to help you learn this language.",
```
Comment (rupali-codes) [inline database/languages/c-programming.json:3]:
```suggestion
    "description": "Learn C Programming  using the book by Kernighan and Ritchi. This free C programming course offers interactive lessons and coding challenges to help you learn this language.",
```

**Comment by CBID2** [comment]
Hey @sakshimishra19! :) I committed @rupali-codes' suggestions to speed up the process! :) @rupali-codes, I think this one is ready to merge! :)

**Comment by ujjawaltyagii** [comment]
Do we really need to commit this ?

image

**Comment by CBID2** [comment]
> Do we really need to commit this ?
> 
> image

Why can't we @ujjawaltyagii? I'm curious. 🤔

**Review by rupali-codes — CHANGES_REQUESTED** [review]
please undo your changes in pnpm-lock.yaml file :)

**Comment by sakshimishra19** [comment]
Hey @rupali-codes 
I have updated the pnpm-lock.yaml file . Kindly review it.

**Review by CBID2 — DISMISSED** [review]
It looks better now! :) LGTM! :)

**Review by rupali-codes — COMMENTED** [review]
Please _uncommit_ your changes in _pnpm-lock.yaml_

**Comment by k-deepak04** [comment]
@sakshimishra19 please resolve the conflicts

**Comment by rupali-codes** [comment]
@CBID2 please be careful while resolving conflicts 😄

**Review by rupali-codes — APPROVED** [review]
lgtm

**Review by CBID2 — APPROVED** [review]
LGTM

**Pull request merged**